assignment.js

```diff
@@ -0,0 +1,265 @@
+    callback(this[i]);
+  };
+};
+
+let arr = [];
+
+[1, 2, 3, 4].myForEach((el) => {
+  // console.log(el * 2);
+});
+
+// Question 2:
+
+// Array.prototype.myMap = function(callback) {
+//   let resultArr = [];
+//   [this].myForEach((el) => {
+//     resultArr.push(callback(this[el]));
+//   });
+// };
+//
+// let mapped = [2, 4, 6].myMap(el => {
+//   // console.log(el);
+// })
+
+Array.prototype.myMap = function(callback) {
+  let mappedArr = [];
+  for (let i = 0; i < this.length; i++) {
+    mappedArr.push(callback(this[i], i));
+  };
+  return mappedArr;
+};
+
+let mapped = [7, 6, 5, 4].myMap(el => {
+  return el*2;
+})
+
+// console.log(mapped);
+
+
+// ** Failing with forEach TToTT ** //
+// Question 3:
+
+// Array.prototype.myFilter = function(callback) {
+//   let filteredArr = [];
+//   [this].myForEach(el => {
+//     if (callback(this[el])) {
+//       filteredArr.push(this[el]);
+//     }]]
+//   })
+//   return filteredArr;
+// }
+
+Array.prototype.myFilter = function(callback) {
+  let myFilterArr = [];
+
+  for (let i = 0; i < this.length; i++) {
+    if (callback(this[i], i)) {
+      myFilterArr.push(this[i]);
+    }
+  }
+  return myFilterArr;
+}
+
+let odds = [1, 2, 3, 4, 5].myFilter(el => {
+  return el % 2;
+  // if (el % 2) {
+  //   return el;
+  // }
+})
+// console.log(odds);
+
+
+
+// Question 4:
+
+// Array.prototype.myEvery = function(callback) {
+//   [this].myForEach(el => {
+//     if (callback(this[el])) {
+//       return true;
+//     } else {
+//       return false;
+//     }
+//   })
+// }
+//
+
+// ** Had it ... then lost it by messing around with it. LOL TT.TT **
+
+Array.prototype.myEvery = function(callback) {
+  for (let i = 0; i < this.length; i++) {
+    if (!(callback(this[i]))) {
+      return false;
+    }
+  }
+}
+
+let randNumArr = [22, 33, 43, 20, 55, 67];
+
+const greaterThan10 = (num) => {
+  return num > 10;
+}
+
+let meh = randNumArr.myEvery(greaterThan10);
+
+// console.log(meh);
+
+
+// Question 5:
+
+Array.prototype.myReduce = function(callback, startVal) {
+  let startIdx = 0;
+  if (!startVal) {
+    startVal = this[0];
+    startIdx++;
+  }
+  for (let i = startIdx; i < this.length; i++) {
+    startVal = callback(startVal, this[i]);
+  }
+  return startVal;
+}
+
+let reduced = [1, 2, 3, 4].myReduce((el, currEl) => {
+  return el + currEl;
+})
+// console.log(reduced);
+
+// Question 6:
+
+Array.prototype.myTranspose = function(callback) {
+  let grid = [];
+
+  for (let i = 0; i < this.length; i++) {
+    grid[i] = [];
+    for (let j = 0; j < this.length; j++) {
+      grid[i][j] = callback(this[j][i]);
+    }
+  }
+  return grid;
+}
+
+let mtx = [
+  [1, 2, 3],
+  [4, 5, 6],
+  [7, 8, 9]
+];
+
+let ehh = mtx.myTranspose(el => {
+  return el;
+})
+
+// console.log(ehh);
+
+// Question 7:
+
+let thisMap = [1, 2, 3, 4, 5].myMap(el => {
+  return el + 10;
+})
+// console.log(thisMap);
+
+// Question 8:
+// **Forgot how to convert an el to an empty string ** //
```

Comment:
ah! Okay, so all you'd have to do is return "" in myMap - remember, the return value is automatically 'mapped' onto the element in the loop!